account_invoice_facturx/models/account_move.py

```diff
@@ -0,0 +1,901 @@
+            )
+            gross_price_amount.text = "%0.*f" % (ns["price_prec"], gross_price_val)
+            fc_discount = float_compare(
+                iline.discount, 0.0, precision_digits=ns["disc_prec"]
+            )
+            if fc_discount in [-1, 1]:
+                trade_allowance = etree.SubElement(
+                    gross_price, ns["ram"] + "AppliedTradeAllowanceCharge"
+                )
+                charge_indic = etree.SubElement(
+                    trade_allowance, ns["ram"] + "ChargeIndicator"
+                )
+                indicator = etree.SubElement(charge_indic, ns["udt"] + "Indicator")
+                if fc_discount == 1:
+                    indicator.text = "false"
+                else:
+                    indicator.text = "true"
+                actual_amount = etree.SubElement(
+                    trade_allowance, ns["ram"] + "ActualAmount"
+                )
+                actual_amount_val = float_round(
+                    gross_price_val - net_price_val, precision_digits=ns["price_prec"]
+                )
+                actual_amount.text = "%0.*f" % (
+                    ns["price_prec"],
+                    actual_amount_val * ns["sign"],
+                )
+
+        net_price = etree.SubElement(
+            line_trade_agreement, ns["ram"] + "NetPriceProductTradePrice"
+        )
+        net_price_amount = etree.SubElement(net_price, ns["ram"] + "ChargeAmount")
+        net_price_amount.text = "%0.*f" % (ns["price_prec"], net_price_val)
+        line_trade_delivery = etree.SubElement(
+            line_item, ns["ram"] + "SpecifiedLineTradeDelivery"
+        )
+        if iline.product_uom_id and iline.product_uom_id.unece_code:
+            unitCode = iline.product_uom_id.unece_code
+        else:
+            unitCode = "C62"
+            if not iline.product_uom_id:
+                logger.warning(
+                    "No unit of measure on invoice line '%s', "
+                    "using C62 (piece) as fallback",
+                    iline.name,
+                )
+            else:
+                logger.warning(
+                    "Missing UNECE Code on unit of measure %s, "
+                    "using C62 (piece) as fallback",
+                    iline.product_uom_id.name,
+                )
+        billed_qty = etree.SubElement(
+            line_trade_delivery, ns["ram"] + "BilledQuantity", unitCode=unitCode
+        )
+        billed_qty.text = "%0.*f" % (ns["qty_prec"], iline.quantity * ns["sign"])
+        line_trade_settlement = etree.SubElement(
+            line_item, ns["ram"] + "SpecifiedLineTradeSettlement"
+        )
+
+        if iline.tax_ids:
+            for tax in iline.tax_ids:
+                self._cii_line_applicable_trade_tax_block(
+                    tax, line_trade_settlement, ns
+                )
+        else:
+            self._cii_line_applicable_trade_tax_block(None, line_trade_settlement, ns)
```

Comment:
Instead of adding an empty tax block, maybe consider doing what I did in https://github.com/OCA/edi/pull/282 when faced a similar issue for UBL generation; case in point is where the company is a VAT-exempt government institution which doesnt bother adding any taxes to invoices since they are exempt from it; but UBL requires the field also, so in that case it picks the default tax. The default tax they select has to be marked with a specific UNECE code that explains _why_ they are tax-exempt, so that code can then be neatly filled in the UNECE field on that default tax.